tests/kafkatest/services/security/security_config.py

```diff
@@ -150,8 +152,16 @@ def __init__(self, context, security_protocol=None, interbroker_security_protoco
             'sasl.kerberos.service.name' : 'kafka'
         }
 
-    def client_config(self, template_props=""):
-        return SecurityConfig(self.context, self.security_protocol, client_sasl_mechanism=self.client_sasl_mechanism, template_props=template_props)
+    def client_config(self, template_props="", static_jaas_conf=True):
+        return SecurityConfig(self.context, self.security_protocol, client_sasl_mechanism=self.client_sasl_mechanism, template_props=template_props, static_jaas_conf=static_jaas_conf)
+
+    def client_config_setup(self, node, template_props=""):
```

Comment:
There is a  comment in `client_config_setup` where `static_jaas_conf` is set that explains why it is set to a different value. `SASL_SSL` is `self.has_sasl and self.has_ssl`. Let me know if the comment needs to be rewritten. Basically, any criteria could have been chosen to set static for some and dynamic for others. I just chose SSL/non-SSL. The intention was to use existing tests to test a new KIP without adding to the time taken for running system tests by modifying some tests to use the new dynamic jaas config.

I used a different `client_config_setup` method for tests which use the dynamic config since they need to be done later in the code when the node is being set up and a second method makes it clearer that a choice of dynamic/static jaas config is being made within the method. But I am ok with changing to a single method if that is better.